### PR TITLE
Add unittests

### LINKED ISSUE #83: Create unit tests
* run tests when building PR or master branch commit

### PR DESCRIPTION
**Pull request by Adman**
* closes #83

Signed-off-by: Adrian Matejov <ado.matejov@gmail.com>

### CODE REVIEW

**Comment by Adman** [comment]
There are some methods in supervisor.py that I'd like to test but it requires that they are moved to referee.py. This will be done in other PR.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -18,3 +18,4 @@ jobs:
       - run: isort --check-only .
       - run: black --check .
       - run: flake8 .
+      - run: pytest
```
[the remark this answers is not in the document]
Comment (Adman) [reply .github/workflows/check.yml]: Added `pytest-cov` package and running the coverage here.

File: controllers/rcj_soccer_referee_supervisor/referee/tests/test_penalty_area_checker.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    return [
+        YELLOW_PENALTY_AREA[0] - 0.001,
+        0.0,
+        YELLOW_PENALTY_AREA[2] - 0.001,
+    ]
+
+
+@pytest.fixture
+def in_blue_penalty_pos() -> List[float]:
+    return [BLUE_PENALTY_AREA[0] + 0.001, 0.0, BLUE_PENALTY_AREA[2] - 0.001]
```
[the remark this answers is not in the document]
Comment (Adman) [reply controllers/rcj_soccer_referee_supervisor/referee/tests/test_penalty_area_checker.py]: Fixed :)

**Comment by Adman** [comment]
@mrshu Note that I introduced managing dependencies using pip-tools. Added some notes about that to README file.

**Comment by mrshu** [comment]
Thanks @Adman, I can get behind `pip-tools`. Maybe even `pipenv` if you decided it would be worth it :slightly_smiling_face:

**Review by mrshu — APPROVED** [review]

**Pull request merged**